devicetypes/fibargroup/fibaro-wall-plug-us-zw5.src/fibaro-wall-plug-us-zw5.groovy

```diff
@@ -0,0 +1,496 @@
+        fingerprint deviceId: "0x1001", inClusters: "0x5E, 0x55, 0x98, 0x9F, 0x22, 0x56, 0x7A, 0x6C", outClusters: "0x86, 0x25, 0x85, 0x8E, 0x59, 0x72, 0x5A, 0x73, 0x32, 0x70, 0x6C, 0x71, 0x75, 0x60"
+
+    }
+
+    tiles (scale: 2) {
+        multiAttributeTile(name:"switch", type: "lighting", width: 3, height: 4, canChangeIcon: true){
+            tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+                attributeState "off", label: 'Off', action: "switch.on", icon: "https://s3-eu-west-1.amazonaws.com/fibaro-smartthings/wallPlugUS/plug_us_off.png", backgroundColor: "#ffffff"
+                attributeState "on", label: 'On', action: "switch.off", icon: "https://s3-eu-west-1.amazonaws.com/fibaro-smartthings/wallPlugUS/plug_us_blue.png", backgroundColor: "#00a0dc"
+            }
+            tileAttribute("device.multiStatus", key:"SECONDARY_CONTROL") {
+                attributeState("multiStatus", label:'${currentValue}')
+            }
+        }
+        valueTile("power", "device.power", decoration: "flat", width: 2, height: 2) {
+            state "power", label:'${currentValue}\nW', action:"refresh"
+        }
+        valueTile("energy", "device.energy", decoration: "flat", width: 2, height: 2) {
+            state "energy", label:'${currentValue}\nkWh', action:"refresh"
+        }
+        valueTile("reset", "device.energy", decoration: "flat", width: 2, height: 2) {
+            state "reset", label:'reset\nkWh', action:"reset"
+        }
+    }
+
+    preferences {
+
+        input (
+                title: "Fibaro Wall Plug manual",
+                description: "Tap to view the manual.",
+                image: "https://s3-eu-west-1.amazonaws.com/fibaro-smartthings/wallPlugUS/plug_us_blue.png",
+                //url: "http://manuals.fibaro.com/content/manuals/en/FGWPEF-102/FGWPEF-102-EN-A-v2.0.pdf",
+                type: "href",
+                element: "href"
+        )
+
+        parameterMap().each {
+            input (
+                    title: "${it.num}. ${it.title}",
+                    description: it.descr,
+                    type: "paragraph",
+                    element: "paragraph"
+            )
+
+            input (
+                    name: it.key,
+                    title: null,
+                    description: it.defValDescr,
+                    type: it.type,
+                    options: it.options,
+                    range: (it.min != null && it.max != null) ? "${it.min}..${it.max}" : null,
+                    defaultValue: it.def,
+                    required: false
+            )
+        }
+
+        input ( name: "logging", title: "Logging", type: "boolean", required: false )
+    }
+}
+
+def test(options, defValue){
+    def mapToString = "$options"
+    def stringToList = mapToString.split(',').collect{it as String}
+    def result = stringToList.get(0).substring(3)
+    return result
+}
+
+//UI and tile functions
+def on() {
+	log.debug "on"

```

Comment:
Based on this line, you'll need to convert their original submission from spaces to tabs.